test/Conversion/MooreToCore/errors.mlir

```diff
@@ -6,3 +6,13 @@ func.func @invalidType() {
 
   return
 }
+
+// -----
+
+func.func @unsupportedConversion() {
+    %0 = moore.constant_string "Test" : i32
+    // expected-error @below {{conversion result type is not currently supported}}
+    // expected-error @below {{failed to legalize operation 'moore.conversion'}}
+    %1 = moore.conversion %0 : !moore.i32 -> !moore.string
```

Comment:
I think there is a dedicated `moore.int_to_string` for this conversion. Does this get emitted somewhere or is this used on purpose to trigger an error 🤔 ?

Reply:
Oh interesting, this is based on an actual error from sv-tests where such a conversion op appeared (the len() test IIRC), so I guess there's somewhere we're emitting a ConversionOp where it should be an IntToStringOp?